pkg/providers/vsphere/config/template-cp.yaml

```diff
@@ -368,7 +368,7 @@ spec:
 {{- end }}
 {{- if and .registryMirrorConfiguration (eq .format "bottlerocket") }}
       registryMirror:
-        endpoint: {{.registryMirrorConfiguration}}
+        endpoint: {{index .registryMirrorConfiguration "public.ecr.aws"}}
```

Comment:
i would vote to pass an specific value for this to the template, this hides too much logic in the template